vm/src/vm.rs

```diff
@@ -321,7 +321,9 @@ impl VirtualMachine {
         self.check_recursive_call("")?;
         self.frames.borrow_mut().push(frame.clone());
         let result = f(frame);
-        self.frames.borrow_mut().pop();
+        // defer dec frame
+        let popped = self.frames.borrow_mut().pop();
+        drop(popped);
```

Comment:
I don't think this really defers anything

Reply:
It was popped while I am working on `__del__`, so it also worked. Will `let _ = self.frames.borrow_mut().pop()` work same?

Reply:
No, it's really unintuitive but if it's `_` it doesn't store as a variable at all, but`_foo` holds onto it till the end of scope

Reply:
Thanks ,I will fix things again